### PR TITLE
Постернак Алексей. Технология SEQ-MPI. Подсчет числа несовпадающих символов двух строк. Вариант 27

### PR DESCRIPTION
**Pull request by AlekseyPosternack**
## Описание

- **Задача**:  Подсчет числа несовпадающих символов двух строк
- **Вариант**:  27
- **Технология**:  SEQ, MPI
- **Описание**: Разработаны последовательная (SEQ) и параллельная (MPI) реализации алгоритма подсчета различающихся символов в двух строках. SEQ версия выполняет поэлементное сравнение, MPI версия распределяет обработку строк между процессами с использованием таких функций, как MPI_Bcast, MPI_Send, MPI_Recv и MPI_Allreduce. Проведено тестирование на строках длиной 10.000.000 символов, показавшее ускорение  2.22 на 4 процессах с последующим анализом результатов.

---

## Чек-лист

- [x] **Статус CI**: Все CI-задачи (сборка, тесты, генерация отчёта) успешно проходят на моей ветке в моем форке
- [x] **Директория и именование задачи**: Я создал директорию с именем `<фамилия>_<первая_буква_имени>_<короткое_название_задачи>`
- [x] **Полное описание задачи**: Я предоставил полное описание задачи в теле pull request
- [x] **clang-format**: Мои изменения успешно проходят `clang-format` локально в моем форке (нет ошибок форматирования)
- [x] **clang-tidy**: Мои изменения успешно проходят `clang-tidy` локально в моем форке (нет предупреждений/ошибок)
- [x] **Функциональные тесты**: Все функциональные тесты успешно проходят локально на моей машине
- [x] **Тесты производительности**: Все тесты производительности успешно проходят локально на моей машине
- [x] **Ветка**: Я работаю в ветке, названной точно так же, как директория моей задачи (например, `nesterov_a_vector_sum`), а не в `master`
- [x] **Правдивое содержание**: Я подтверждаю, что все сведения, указанные в этом pull request, являются точными и достоверными

### CODE REVIEW

File: tasks/posternak_a_count_different_char_in_two_lines/data/pic.jpg
Comment (aobolensk) [inline tasks/posternak_a_count_different_char_in_two_lines/data/pic.jpg]: Remove unused files

File: tasks/posternak_a_count_different_char_in_two_lines/tests/functional/main.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ public:
+  static std::string PrintTestParam(const TestType &test_param) {
+    return std::to_string(std::get<0>(test_param)) + "_" + std::get<1>(test_param);
+  }
+
+ protected:
+  void SetUp() override {
+    TestType params = std::get<static_cast<std::size_t>(ppc::util::GTestParamIndex::kTestParams)>(GetParam());
+    int test = std::get<0>(params);
+
```
Comment (aobolensk) [inline tasks/posternak_a_count_different_char_in_two_lines/tests/functional/main.cpp]: Please, add more functional tests to cover more corner cases

File: tasks/posternak_a_count_different_char_in_two_lines/report.md [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+
+Итого, для малых объемов данных лучше использовать последовательный (SEQ) алгоритм, а для больших объемов данных - параллельный (MPI) алгоритм.
+
+## 9. Литература
+
+1. Документация по курсу: "Параллельное программирование": https://learning-process.github.io/parallel_programming_course/ru/index.html (Оболенский А.А, Нестеров А.Ю)
+2. Лекции по курсу "Параллельное программирование". (Сысоев А.В. ННГУ 2025 г.)
+3. Документация по MPI: https://www.open-mpi.org/
+
+## Приложение
```
Comment (aobolensk) [inline tasks/posternak_a_count_different_char_in_two_lines/report.md]: If the appendix is just a copy-paste of implementation files, I'd rather remove it

File: tasks/posternak_a_count_different_char_in_two_lines/data/pic.jpg
Comment (AlekseyPosternack) [reply tasks/posternak_a_count_different_char_in_two_lines/data/pic.jpg]: Unused folder "data" removed

File: tasks/posternak_a_count_different_char_in_two_lines/tests/functional/main.cpp (hunk 1 above)
Comment (AlekseyPosternack) [reply tasks/posternak_a_count_different_char_in_two_lines/tests/functional/main.cpp]: Added more functional tests (20 instead of 5)
